app/controllers/games_controller.rb

```diff
@@ -21,6 +21,7 @@ def score
   def game
     # looks for rounds with our game id
     @round = Round.where(game_id: @game.id)
+    # why do we check for presence here and why is round an array on the next line
```

Comment:
Q: Why check for presence, ie. why do we see run `round[0].nil?` 

A: Because the we want to change what we display in the view (and how we prepare info  to pass to the view) depending whether not the round exists.

We only persist rounds (i.e. call `#save` on the round) if the round has an answer. So, we know that if no rounds were found for the `game_id` in the `params` then we need to get a random `question`, and its associated `answers`, and pass them along to the view using the `@question` and `@answers`instance variables execution flows to the view (which explains the next couple of lines).